tasks/main.yml

```diff
@@ -553,188 +568,221 @@
+            state: "{{ 'present' if mssql_ha_configure | bool else 'absent' }}"
+          register: __mssql_server_ha_packages_install
+
+        - name: Enable the hadrenabled setting
+          include_tasks: mssql_conf_setting.yml
+          vars:
+            __mssql_conf_setting: "hadr hadrenabled"
+            __mssql_conf_setting_value: 1
+
+        # flush_handlers task does not support when conditional
+        - name: Restart the mssql-server service if hadrenabled task was changed
+          service:
+            name: mssql-server
+            state: restarted
+          when: (__mssql_conf_set is changed) or
+            (__mssql_server_ha_packages_install is changed)
+          register: __mssql_replica_restarted
 
-    - name: Distribute certificate and private key to managed nodes
-      copy:
-        src: "{{ item.key }}"
-        dest: "{{ item.value }}"
-        owner: mssql
-        group: mssql
-        mode: 0660
-        force: true
-      with_dict:
-        cert: "{{ __mssql_ha_cert_dest }}"
-        key: "{{ __mssql_ha_private_key_dest }}"
+        - name: Remove certificate from SQL Server
+          vars:
+            __mssql_sql_files_to_input:
+              - drop_cert.j2
+          include_tasks: input_sql_files.yml
+          when: mssql_ha_reset_cert | bool
+
+        - name: Distribute certificate and private key to managed nodes
+          copy:
+            src: "{{ item.key }}"
+            dest: "{{ item.value }}"
+            owner: mssql
+            group: mssql
+            mode: 0660
+            force: true
+          with_dict:
+            cert: "{{ __mssql_ha_cert_dest }}"
+            key: "{{ __mssql_ha_private_key_dest }}"
+
+        - name: Configure SQL entities on not primary replicas
+          vars:
+            __mssql_sql_files_to_input:
+              - enable_alwayson.j2
+              - create_master_key_encryption.j2
+              - restore_cert.j2
+              - configure_endpoint.j2
+              - create_ha_login.j2
+              - join_to_ag.j2
+              - grant_permissions_to_ha_login.j2
+          include_tasks: input_sql_files.yml
+      always:
+        # changed_when: false because this task removes remnants of cert files
+        - name: Remove certificate and private key from the control node
+          delegate_to: localhost
+          file:
+            path: "{{ item }}"
+            state: absent
+          loop:
+            - cert
+            - key
+          changed_when: false
 
-    - name: Configure SQL entities on not primary replicas
+    - name: Configure pacemaker
+      block:
+        - name: Save credentials for the {{ mssql_ha_login }} SQL Server login
+          copy:
+            content: |
+              {{ mssql_ha_login }}
+              {{ mssql_ha_login_password }}
+            dest: /var/opt/mssql/secrets/passwd
+            owner: root
+            group: root
+            mode: 0400
+            force: true
```

Comment:
should have a `no_log: true` here

Reply:
It should, I should check the role for security everywhere. Add a test with the new vault CI feature

Reply:
Diff in this PR is very dirty so I will merge it and then continue in future PRs

Reply:
Fixing in  #119